app/src/main/java/com/jerboa/ui/components/community/list/CommunityListActivity.kt

```diff
@@ -53,24 +56,96 @@ fun CommunityListActivity(
                     search = search,
                     onSearchChange = {
                         search = it
-                        fetchCommunitiesJob?.cancel()
-                        fetchCommunitiesJob = scope.launch {
-                            delay(DEBOUNCE_DELAY)
-                            communityListViewModel.searchCommunities(
-                                form = Search(
-                                    q = search,
-                                    type_ = SearchType.Communities,
-                                    sort = SortType.TopAll,
-                                    auth = account?.jwt,
-                                ),
-                            )
+                        if (it.isEmpty()) {
+                            communityListViewModel.resetSearch()
+                            return@CommunityListHeader
+                        }
+                        scope.launch {
+                            communityListViewModel.searchAllCommunities(search, account?.jwt, true)
                         }
                     },
                 )
             },
             content = { padding ->
                 when (val communitiesRes = communityListViewModel.searchRes) {
-                    ApiState.Empty -> ApiEmptyText()
+                    ApiState.Empty -> {
+                        Column(
+                            modifier = Modifier
+                                .padding(padding)
+                                .imePadding(),
+                        ) {
+                            val history by communityListViewModel.searchHistory.collectAsState(
+                                emptyList(),
+                            )
+                            if (history.isNotEmpty()) {
+                                ListItem(
+                                    headlineContent = {
+                                        Text(
+                                            text = "Recent searches",
```

Comment:
Strings in this file need to be localized, then LGTM

Reply:
Added string resources.